internal/services/instance_types_service.go

```diff
@@ -61,8 +63,38 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := render.RenderList(w, r, payloads.NewListInstanceTypeResponse(&res)); err != nil {
+	instances := convertAWSTypes(res)
+	if err != nil {
+		renderError(w, r, payloads.NewAWSError(r.Context(), "can't convertAWSTypes", err))
+		return
+	}
+
+	filteredInstances, err := supported_types.FilterAWSInstanceTypes(r.Context(), &instances)
+	if err != nil {
+		renderError(w, r, payloads.NewAWSError(r.Context(), "can't FilterUnsupportedTypes", err))
+		return
+	}
+	if err := render.RenderList(w, r, payloads.NewListInstanceTypeResponse(filteredInstances)); err != nil {
 		renderError(w, r, payloads.NewRenderError(r.Context(), "list instance types", err))
 		return
 	}
 }
+
+func convertAWSTypes(types []types.InstanceTypeInfo) []clients.InstanceType {
+	list := make([]clients.InstanceType, 0, len(types))
+	for i := range types {
+		arch := make([]string, 0, len(types[i].ProcessorInfo.SupportedArchitectures))
+		for _, a := range types[i].ProcessorInfo.SupportedArchitectures {
+			arch = append(arch, string(a))
+		}
+		list = append(list, clients.InstanceType{
+			Name:                   string(types[i].InstanceType),
+			VCPUs:                  types[i].VCpuInfo.DefaultVCpus,
+			Cores:                  types[i].VCpuInfo.DefaultCores,
+			Memory:                 *types[i].MemoryInfo.SizeInMiB,
+			SupportedArchitectures: arch,
+			Supported:              false,
+		})
+	}
+	return list
+}
```

Comment:
I suggest to move this function next to `FilterAWSInstanceTypes` and name it in Go way: `NewInstanceTypes` since it is really a "constructor" type of function.